### PR TITLE
eip2930: add payload methods for `TxEip2930`

### PR DESCRIPTION
**Pull request by tcoratger**
Related https://github.com/paradigmxyz/reth/pull/10623

### CODE REVIEW

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
looks like we need to bump msrv

**Comment by tcoratger** [comment]
> looks like we need to bump msrv

@mattsse done here https://github.com/alloy-rs/alloy/pull/1219

File: crates/consensus/src/transaction/eip2930.rs [hunk 1]
```diff
@@ -223,6 +223,19 @@ impl TxEip2930 {
...
+
+    /// Output the length of the RLP signed transaction encoding, _without_ a RLP string header.
+    pub fn payload_len_with_signature_without_header(&self, signature: &Signature) -> usize {
+        let payload_length = self.fields_len() + signature.rlp_vrs_len();
+        // 'transaction type byte length' + 'header length' + 'payload length'
+        1 + length_of_length(payload_length) + payload_length
+    }
+
+    /// Output the length of the RLP signed transaction encoding. This encodes with a RLP header.
+    pub fn payload_len_with_signature(&self, signature: &Signature) -> usize {
```
Comment (mattsse) [inline crates/consensus/src/transaction/eip2930.rs:235]: should all of those accept `S: EncodableSignature` instead?

**Review by mattsse — COMMENTED** [review]
ptal @klkvr

File: crates/consensus/src/transaction/eip2930.rs (hunk 1 above)
Comment (tcoratger) [reply crates/consensus/src/transaction/eip2930.rs]: Same doesn't hurt to make the methods more generic imo, just modified

**Comment by klkvr** [comment]
hmm, not sure we need those? we already have `encoded_len_with_signature` and `encode_with_signature` which accept `with_header: bool`

**Comment by tcoratger** [comment]
Oh that's true thanks

**Pull request closed without merging**